### ISSUE TITLE
false positive `Function "_foo" is not accessed `

### ISSUE THREAD

**Issue by dybi** [microsoft/pyright#6747]
**Describe the bug**
`pyright` seems to give false positive results for `reportUnusedFunction` when a function is "private" (starts with `_`) and it's only usages are outside the module where it is defined. Please see the code snippets for details.

`pyright` reports: `error: Function "_foo" is not accessed (reportUnusedFunction)` which is not true. 

**Code or Screenshots**
`main.py`
```python
from functions import _foo

def bar(x):
    return _foo(x)

if __name__ == '__main__':
    bar(3)
```

`functions.py`
```python
def _foo(x):
    return x * 2
```

**VS Code extension or command-line**
running pyright from CLI
ver 1.1.340

`pyproject.toml`
```toml
[tool.pyright]

pythonVersion = "3.11"
pythonPlatform = "Linux"
typeCheckingMode = "basic"

reportUnusedFunction = true

```

**Comment by dybi** [comment]
Look, I am trying to introduce pyright to a large, legacy code base. It has it's hairy places and I have been counting on pyright to help finding them.
I have filed this issue as I found `Function "_foo" is not accessed (reportUnusedFunction)` error report misleading.
